torch/testing/_internal/distributed/rpc/rpc_test.py

```diff
@@ -1055,6 +1055,35 @@ def test_all_gather_timeout(self):
             with self.assertRaisesRegex(RuntimeError, expected_error):
                 rpc.api._all_gather(SlowPickleClass(0.5))
 
+    @dist_init
+    def test_rpc_barrier_all(self):
+        info = rpc.get_worker_info()
+        all_worker_info = rpc._get_current_rpc_agent().get_worker_infos()
+        names = [worker.name for worker in all_worker_info]
+        rpc.api._barrier(names)
```

Comment:
Ideally a test should assert something to test the functionality, though admittedly it is a bit hard for barrier. Could we do something like, before the barrier, assert some global count is 0, then send RPC to each node to increment the global count, and then call barrier, then assert the count == world_size? This will ensure barrier properly waits for all outstanding work. Open to better ideas though.

Reply:
That's a good point. I'll implement something along the count idea. I'll also add something to test the negative case where barrier does actually time out.